### DISCUSSION TITLE
Add npm package to filament and import it

### DISCUSSION THREAD

**Discussion by FrazeColder** [filamentphp/filament#9972] [Help]
Hey there,

I am quite new to Filament PHP and I have to say it's such an awesome tool. However, Livewire and Alpine.js is also quite new for me, therefore I have a "silly question". 

I want to create my own custom form field with QuillJs. I have already run the command `php artisan make:form-field QuillEditor` which has generated me the `QuillEditor.php` class and my `quill-editor.blade.php` blade file.

However, at the moment I am using a CDN to include my `.js` and `.css` files. I want to get rid of that and import Quill via my npm modules. 

I have no experience with Laravel Mix or Laravel Vite and also no `app.js` so far in my resources folder. Can you guys please help me and tell me what I have to do in order to import Quill in my `quill-editor.blade.php` and also how I can import and make Quill register modules such as [quill-image-uploader](https://www.npmjs.com/package/quill-image-uploader)? 

This is my code so far I want to change:
```
<script src="https://cdn.quilljs.com/1.3.6/quill.js"></script>
<link href="https://cdn.quilljs.com/1.3.6/quill.snow.css" rel="stylesheet">

<x-dynamic-component
    :component="$getFieldWrapperView()"
    :field="$field"
>
    <div
        x-data="{ state: $wire.$entangle('{{ $getStatePath() }}') }"
        x-init="
            const quill = new Quill($refs.editor, {
                theme: 'snow'
            });

            quill.setText(state);

            quill.on('text-change', () => {
                console.log('get html', quill.root.innerHTML);
                state = quill.root.innerHTML;
            });
    "
    >
        <div
            wire:ignore
            x-ref="editor"
        ></div>
    </div>
</x-dynamic-component>

```

Kind regards

**Comment by danharrin** [comment]
It’ll be in the “Async Alpine” episode of Laracasts I think. Let me know if not.
